modules/util/portal-tools-service-builder/src/main/resources/com/liferay/portal/tools/service/builder/dependencies/persistence_test.ftl

```diff
@@ -330,6 +330,222 @@ public class ${entity.name}PersistenceTest {
 	}
 
 	<#if entity.hasExternalReferenceCode()>
+		<#if entity.versionEntity??>
+			@Test(expected = ${duplicateEntityExternalReferenceCode}Exception.class)
+			public void testUpdateAddDraftWithExistingExternalReferenceCode()
+				throws Exception {
+				${entity.name} ${entity.variableName} = add${entity.name}();
+
+				addDraftForExisting${entity.name}(${entity.variableName});
+
+				addDraftForExisting${entity.name}(${entity.variableName});
+			}
+
+			@Test
+			public void testUpdateAddDraft()throws Exception {
+				${entity.name} ${entity.variableName} = add${entity.name}();
+
+				${entity.name} draft${entity.name} = addDraftForExisting${entity.name}(${entity.variableName});
+
+				<#list entity.regularEntityColumns as entityColumn>
+					<#if !entityColumn.primary && (validator.isNull(parentPKColumn) || (parentPKColumn.name != entityColumn.name))>
+						<#if stringUtil.equals(entityColumn.name, "mvccVersion")>
+
+						<#elseif stringUtil.equals(entityColumn.name, "headId")>
+							Assert.assertEquals(${entity.variableName}.get${entityColumn.methodName}(), -draft${entity.name}.get${entityColumn.methodName}());
+						<#elseif stringUtil.equals(entityColumn.name, "status")>
+							Assert.assertEquals(2, draft${entity.name}.get${entityColumn.methodName}());
+						<#elseif stringUtil.equals(entityColumn.type, "Blob")>
+							Blob ${entityColumn.methodName} = ${entity.variableName}.get${entityColumn.methodName}();
+							Blob draft${entityColumn.methodName} = draft${entity.name}.get${entityColumn.methodName}();
+
+							Assert.assertTrue(
+								Arrays.equals(${entityColumn.variableName}.getBytes(1, (int)${entityColumn.variableName}.length()),
+								Arrays.equals(draft${entityColumn.name}.getBytes(1, (int)draft${entityColumn.name}.length()));
+						<#elseif stringUtil.equals(entityColumn.type, "boolean")>
+							Assert.assertEquals(${entity.variableName}.is${entityColumn.methodName}(), draft${entity.name}.is${entityColumn.methodName}());
+						<#elseif stringUtil.equals(entityColumn.type, "Date")>
+							Assert.assertEquals(Time.getShortTimestamp(${entity.variableName}.get${entityColumn.methodName}()), Time.getShortTimestamp(draft${entity.name}.get${entityColumn.methodName}()));
+						<#elseif stringUtil.equals(entityColumn.type, "double")>
+							AssertUtils.assertEquals(${entity.variableName}.get${entityColumn.methodName}(), draft${entity.name}.get${entityColumn.methodName}());
+						<#else>
+							Assert.assertEquals(${entity.variableName}.get${entityColumn.methodName}(), draft${entity.name}.get${entityColumn.methodName}());
+						</#if>
+					</#if>
+				</#list>
+
+			}
+
+			@Test(expected = ${duplicateEntityExternalReferenceCode}Exception.class)
+			public void testUpdateAddWithExistingExternalReferenceCodeHead()
```

Comment:
What do you mean exactly with this? 

The tests that have been added cover the following:
- You can't create a new FragmentEntry with an already existing ERC.
- You can create a new DraftFragmentEntry with an existing ERC for an already existing entry.
- You can't create a new DraftFragmentEntry if there's already another DraftFragmentEntry with the same ERC.

I know that because of the way it's implemented, the last case always works if the first one also works but I've added it just in case this implementation changes in the future but I don't know if this is what you are trying to say...

Reply:
As we talked, we are already testing with the same ERC and head value